src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -110,13 +122,25 @@ bool IDictionary.IsReadOnly
 			get { return arguments.IsReadOnly; }
 		}
 
+		public static Arguments Empty { get; } = new Arguments();
+
 		public void Add(object key, object value)
 		{
+			if (isReadOnly)
+			{
+				throw new NotSupportedException("You are trying to modify a read-only dictionary");
```

Comment:
.NET throws:
`System.NotSupportedException: Collection is read-only.`

You don't want the "you are" otherwise the exception reads strange.

Maybe also worth just having an `EnsureWritable` or something and only have the message once.

Reply:
Will fix.